src/index.ts

```diff
@@ -310,4 +310,182 @@ export namespace OpenAI {
   export import FunctionParameters = API.FunctionParameters;
 }
 
+// ---------------------- Azure ----------------------
+
+/** API Client for interfacing with the Azure OpenAI API. */
+export interface AzureClientOptions extends ClientOptions {
+  /**
+   * Defaults to process.env['OPENAI_API_VERSION'].
+   */
+  apiVersion?: string | undefined;
+  /**
+   * Your Azure endpoint, including the resource, e.g. `https://example-resource.azure.openai.com/`
+   */
+  endpoint?: string | undefined;
+  /**
+   * A model deployment, if given, sets the base client URL to include `/deployments/{deployment}`.
+   * Note: this means you won't be able to use non-deployment endpoints. Not supported with Assistants APIs.
+   */
+  deployment?: string | undefined;
+  /**
+   * Defaults to process.env['AZURE_OPENAI_API_KEY'].
+   */
+  apiKey?: string | undefined;
+  /**
+   * A function that returns a Microsoft Entra (formerly known as Azure Active Directory) access token, will be invoked on every request.
+   */
+  microsoftEntraTokenProvider?: (() => string) | undefined;
```

Comment:
is not supporting the equivalent `azure_ad_token` as well like in the [Python client](https://github.com/openai/openai-python/blob/aed1e43745cd6358b4bafd3a39b3dfeee5e31a03/src/openai/lib/azure.py#L166) an intentional mismatch? (I haven't read all the context)

in other words, would people be suprised they can't set the `AZURE_OPENAI_AD_TOKEN` environment variable and have it picked up here?

Reply:
@deyaaeldeen in addition to Robert's comments, in the Python client I believe we call this `azure_ad` not `microsoft_entra` – is that correct? Can we keep the naming consistent and call it Azure Active Directory instead?

Reply:
> is not supporting the equivalent azure_ad_token as well like in the [Python client](https://github.com/openai/openai-python/blob/aed1e43745cd6358b4bafd3a39b3dfeee5e31a03/src/openai/lib/azure.py#L166) an intentional mismatch?

Yes, I would like our customers to use the token provider first because it takes care of token refreshing. I can add `azureAdToken` in the future if it turns out to be needed.

>  Can we keep the naming consistent and call it Azure Active Directory instead?

Ok I made the update in https://github.com/openai/openai-node/pull/718/commits/670165f52c4599f8406364b9778646ebb5a131ab and we can plan the switch to the Entra name in the next major version.